tests/e2e_vaultwarden_test.py

```
@@ -104,15 +183,71 @@ def main() -> None:
+        logger.info("Logging in to Bitwarden")
+        initial_session = _login_session(env, bw_email, bw_password)
+        logger.info("Listing pre-migration items")
         before_items = _bw_json(env, "list", "items", "--session", initial_session)
+        logger.info(f"Found {len(before_items)} existing items")
+
+        # Diagnostic: smoke-test bw serve directly to verify it can start
+        logger.info("Smoke-testing bw serve directly")
+        serve_env = env.copy()
+        serve_env["BW_SESSION"] = initial_session
+        serve_proc = subprocess.Popen(
+            ["bw", "serve", "--port", "19876", "--hostname", "localhost"],
```

Ruff (0.15.2)

[error] Starting a process with a partial executable path

(S607)